analyzer/config/hcl2/hcl2_test.go

```diff
@@ -0,0 +1,108 @@
+package hcl2
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/open-policy-agent/conftest/parser/hcl2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/analyzer/config"
+	"github.com/aquasecurity/fanal/types"
+)
+
+func Test_hcl2ConfigAnalyzer_Analyze(t *testing.T) {
+	tests := []struct {
+		name      string
+		inputFile string
+		want      *analyzer.AnalysisResult
+		wantErr   string
+	}{
+		{
+			name:      "happy path",
+			inputFile: "testdata/deployment.hcl",
+			want: &analyzer.AnalysisResult{
+				Configs: []types.Config{
+					{
+						Type:     config.HCL2,
+						FilePath: "testdata/deployment.hcl",
+						Content: map[string]interface{}{
+							"apiVersion": "apps/v1",
+							"kind":       "Deployment",
+							"metadata": map[string]interface{}{
+								"name": "hello-kubernetes",
+							},
+							"spec": map[string]interface{}{
+								"replicas": float64(3),
```

Comment:
[conftest hcl2 parser](https://github.com/open-policy-agent/conftest/tree/master/parser/hcl2) does not use [hashicorp's library](https://github.com/hashicorp/hcl), which results in this inaccurate unmarshalling. for our purpose, i think it does not really matter, but do you think we should try to use the native library instead?

Reply:
What kind of differences are there? `float64(3)` looks fine.

Reply:
`hcl2json` seems to use `hashicorp/hcl/v2` internally.
https://github.com/tmccombs/hcl2json/blob/main/convert/convert.go#L8

Reply:
it's just that given this HCL config
```
spec {
  replicas = 3
}
```
the current parser will give
```
"spec": map[string]interface{}{
  "replicas": float64(3),
},
```
instead of
```
"spec": map[string]interface{}{
  "replicas": int(3), // or any other int type
},
```
which is probably not a big deal for what we're trying to do

Reply:
I'm wondering it is difference between hashicorp/hcl/v1 and hashicorp/hcl/v2, not between tmccombs/hcl2json/convert and hashicorp/hcl/v2.

Reply:
If `tmccombs/hcl2json/convert` results in inaccurate results, we should replace it with `hashicorp/hcl/v2`, but the library internally uses `hashicorp/hcl/v2`. So, I feel like `float64(3)` is a correct result. Could you parse the test file by `hashicorp/hcl/v2` and check if it returns int(3) or float64(3)?

Reply:
FWIW, the spec mentions [this](https://github.com/hashicorp/hcl/blob/main/json/spec.md#numbers) and [this](https://github.com/hashicorp/hcl/blob/main/spec.md#primitive-types)

Reply:
Hmm. I wanted to look into it, but it doesn't matter in our case actually. If we find an issue in the future, we can fix it.